File: sdks/csharp/src/NovaBilling/Portal/Requests/GetInvoicesPortalRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record GetInvoicesPortalRequest
{
    /// <summary>
    /// Customer external ID
    /// </summary>
    [JsonIgnore]
    public required string ExternalId { get; set; }

    [JsonIgnore]
    public GetInvoicesPortalRequestStatus? Status { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Portal/Requests/GetInvoicesPortalRequest.php
<?php

namespace NovaBilling\Portal\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Portal\Types\GetInvoicesPortalRequestStatus;

class GetInvoicesPortalRequest extends JsonSerializableType
{
    /**
     * @var ?value-of<GetInvoicesPortalRequestStatus> $status
     */
    public ?string $status;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   status?: ?value-of<GetInvoicesPortalRequestStatus>,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->status = $values['status'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}
